### PR TITLE
[Magiclysm] Add Magiclysm academy

### PR DESCRIPTION
**Pull request by curstwist**
#### Summary
 ```SUMMARY: Mods "Adds new magic academy city building, maps for lairs, retreats."```

#### Purpose of change
A common concern hear for magiclysm is how hard it can be to get scrolls for a specific class.  This academy aims to lessen that problem, and also offers maps for the harder to find locations.

Also added a photograph hint in the magic basement to improve people's chances of finding the door trigger.

#### Describe the solution
 In adding this academy, I've made loots fairly generous (may need adjustment) and I've set up item_groups for class specific loots (and classless).  Each student chamber represents a student specializing in one of the 8 classes and should be a reliable place to find scrolls and occasional books.

The two teacher apartments are more general.  There is also an alchemy lab with a good amount of supplies, 2 class rooms and a small library.  

I've added 2 maps, one for lairs and one advertising the academy's retreat locations (wizard lake retreat and magic cabins).

#### Describe alternatives you've considered
Less loot, but I'd like some feed back on how this is for players first.  
Right now it has golums and zombie spawns, I may eventually increase the depth and add some monster hunting/training grounds.  I need to think about what I want to do and didn't want to sit on this part any longer.

#### Testing
Spawned in world, tested maps, tested spawns, lots of loot checking.

#### Additional context
The screenshots:
ground:
academy_ground

basement:
academy_basement

2nd:
academy_2nd

3rd
academy_3rd

4th
academy_4th

5th
academy_5th

6th
academy_6th

7th
academy_7th

8th
academy_8th

roof:
academy_roof

### CODE REVIEW

File: data/mods/Magiclysm/items/books_lore.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      "message": "You add the locations to your map."
+    }
+  },
+  {
+    "id": "wizard_photo",
+    "type": "GENERIC",
+    "//": "Unique item for magic_basement.",
+    "category": "other",
+    "name": "old photo",
+    "description": "A photo of an jovial, old wizard, he seems to be dancing with a coat rack in this basement.  There is a stack of suitcases in the background.",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/books_lore.json]:
```suggestion
    "description": "A photo of a jovial, old wizard, he seems to be dancing with a coat rack in this basement.  There is a stack of suitcases in the background.",
```

File: data/mods/Magiclysm/items/spell_scrolls.json [hunk 2]
```diff
@@ -662,6 +743,7 @@
     "type": "GENERIC",
     "copy-from": "spell_scroll",
     "id": "spell_scroll_lava_bomb",
+    "//": "Classless spell",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/spell_scrolls.json]: if lava bomb is classless, that's a mistake, it should be earthshaper.

**Comment by KorGgenT** [comment]
sweet. just gotta spawn in a school and check it out; the non map related json changes LGTM

File: data/mods/Magiclysm/items/spell_scrolls.json (hunk 2 above)
Comment (curstwist) [reply data/mods/Magiclysm/items/spell_scrolls.json]: image

Comment (curstwist) [reply data/mods/Magiclysm/items/spell_scrolls.json]: not sure how to change that UI display but I changed my note and its new item_group

Comment (KorGgenT) [reply data/mods/Magiclysm/items/spell_scrolls.json]: oh, maybe the spell itself got missed then

**Comment by KorGgenT** [comment]
i spawned in an academy and i started seeing the class runes. those shouldn't spawn as they are representations of your class that allow you to do certain crafting recipes, as limiting crafting recipes by trait isn't something i wanted to do.

**Pull request merged**